docs/samples/Microsoft.ML.Samples/Dynamic/Transforms/Concatenate.cs

```diff
@@ -1,64 +1,78 @@
+            // A pipeline for concatenating the "Feature1", "Feature2" and "Feature3" columns together into a vector that will be the Features column.
             // Concatenation is necessary because learners take **feature vectors** as inputs.
             //   e.g. var regressionTrainer = mlContext.Regression.Trainers.FastTree(labelColumn: "Label", featureColumn: "Features");
-            string outputColumnName = "Features";
-            var pipeline = mlContext.Transforms.Concatenate(outputColumnName, new[] { "Age", "Parity", "Induced" });
+            //
+            // Please note that the "Feature3" column is converted from int32 to float using the ConvertType API.
+            // The Concatenate API requires all the columns to be of same type.
+            var pipeline = mlContext.Transforms.Conversion.ConvertType("Feature4", "Feature3", outputKind: DataKind.Single)
+                .Append(mlContext.Transforms.Concatenate("Features", new[] { "Feature1", "Feature2", "Feature4" }));
 
             // The transformed data.
-            var transformedData = pipeline.Fit(trainData).Transform(trainData);
+            var transformedData = pipeline.Fit(dataview).Transform(dataview);
 
             // Now let's take a look at what this concatenation did.
-            // We can extract the newly created column as an IEnumerable of SampleInfertDataWithFeatures, the class we define above.
-            var featuresColumn = mlContext.Data.CreateEnumerable<SampleInfertDataWithFeatures>(transformedData, reuseRowObject: false);
+            // We can extract the newly created column as an IEnumerable of TransformedData.
+            var featuresColumn = mlContext.Data.CreateEnumerable<TransformedData>(transformedData, reuseRowObject: false);
 
             // And we can write out a few rows
-            Console.WriteLine($"{outputColumnName} column obtained post-transformation.");
+            Console.WriteLine($"Features column obtained post-transformation.");
             foreach (var featureRow in featuresColumn)
             {
-                foreach (var value in featureRow.Features.GetValues())
+                foreach (var value in featureRow.Features)
                     Console.Write($"{value} ");
                 Console.WriteLine("");
             }
 
             // Expected output:
             // Features column obtained post-transformation.
             //
-            // 26 6 1
-            // 42 1 1
-            // 39 6 2
-            // 34 4 2
-            // 35 3 1
+            //   0.1 1.1 2.1 3.1 1
+            //   0.2 1.2 2.2 3.2 2
+            //   0.3 1.3 2.3 3.3 3
+            //   0.4 1.4 2.4 3.4 4
+            //   0.5 1.5 2.5 3.5 5
+            //   0.6 1.6 2.6 3.6 6
+        }
+
+        private class InputData
+        {
+            public float Feature1;
+            public float[] Feature2;
```

Comment:
I would put VectorType(3) here.
Otherwise you concat scalar+VarVector+Scalar  which would be VarVector and you can't use it for trainers.
Or maybe include that in sample, two concats, one on varvector, one on vector....